### PR TITLE
[ Doc ] Add Documentation of MNIST Application

### PR DESCRIPTION
**Pull request by jijoongmoon**
Add README.md and imagews for mnist.

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: Applications/MNIST/README.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
+# MNIST
+
+In this example, we demostrate training full neural network model with mnist dataset.
```
Comment (wooksong) [inline Applications/MNIST/README.md:3]:
```suggestion
In this example, we demonstrate training full neural network model with mnist dataset.
```

File: Applications/MNIST/README.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+...
+0 0 0 223 225 0 0 0 0 0 0... 0  --> ith feature data (784xsizeof(float))
+1 0 0 0 0 0 0 0 0 0         --> ith label data as an one-hot vector
+0 0 0 0 0 0 0 0 224 123 0... 0  --> i+1th feature data (784xsizeof(float))
+0 1 0 0 0 0 0 0 0 0         --> i+1th label data as an one-hot vector
+...
+
+```
+
+In order to get the training data, ```DataBufferFromCallback``` option is used and ```getData_train``` and ```getData_val``` function are defined which reads the feature and label with batch_size and fill the databuffer. If there is no data for a epoch, then ```last``` will be true.
```
Comment (wooksong) [inline Applications/MNIST/README.md:26]:
```suggestion
In order to get the training data, ```DataBufferFromCallback``` option is used and ```getData_train``` and ```getData_val``` functions are defined which reads the feature and label with batch_size and fill the databuffer. If there is no data for an epoch, then ```last``` will be true.
```

File: Applications/MNIST/README.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+ * @param[out] outLabel
+ * @param[out] last if the data is finished
+ * @param[in] user_data private data for the callback
+ * @retval status for handling error
+ */
+int getBatch_train(float **outVec, float **outLabel, bool *last,
+                   void *user_data);
+```
+
+### MNIST Configuraion file
```
Comment (wooksong) [inline Applications/MNIST/README.md:41]:
```suggestion
### MNIST Configuration file
```

File: Applications/MNIST/README.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+The model configuraion is in ```res/mnist.ini```. ADAM optimizer is used and the trained model is going to be saved in ```model.bin```.
```
Comment (wooksong) [inline Applications/MNIST/README.md:42]:
```suggestion
The model configuration is in ```res/mnist.ini```. ADAM optimizer is used and the trained model is going to be saved in ```model.bin```.
```

Comment (jijoongmoon) [reply Applications/MNIST/README.md]: fixed. thanks.

File: Applications/MNIST/README.md (hunk 3 above)
Comment (jijoongmoon) [reply Applications/MNIST/README.md]: fixed. thanks.

File: Applications/MNIST/README.md (hunk 2 above)
Comment (jijoongmoon) [reply Applications/MNIST/README.md]: fixed. thanks.

File: Applications/MNIST/README.md (hunk 1 above)
Comment (jijoongmoon) [reply Applications/MNIST/README.md]: fixed. thanks.

File: Applications/MNIST/README.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+```dataset.py``` is for the data generation from the input data file, ```mnist_trainingSet.dat```.
+
+
+You can run with
+
+``` bash
+$ python3 Training_Keras.py
+```
+
+```mnist_trainingSet.data``` must be in the same directory with ```Training_Kears.py```.
```
Comment (zhoonit) [inline Applications/MNIST/README.md:128]:
```suggestion
```mnist_trainingSet.dat``` must be in the same directory with ```Training_Kears.py```.
```

**Review by zhoonit — APPROVED** [review]
LGTM!

File: Applications/MNIST/README.md (hunk 5 above)
Comment (jijoongmoon) [reply Applications/MNIST/README.md:128]: File name is mnist_trainigSet.dat.

File: Applications/MNIST/README.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+...
+0 0 0 223 225 0 0 0 0 0 0... 0  --> ith feature data (784xsizeof(float))
+1 0 0 0 0 0 0 0 0 0         --> ith label data as an one-hot vector
+0 0 0 0 0 0 0 0 224 123 0... 0  --> i+1th feature data (784xsizeof(float))
+0 1 0 0 0 0 0 0 0 0         --> i+1th label data as an one-hot vector
+...
+
+```
+
+In order to get the training data, ```DataBufferFromCallback``` option is used and ```getData_train``` and ```getData_val``` functions are defined which reads the feature and label with batch_size and fill the databuffer. If there is no data for an epoch, then ```last``` will be true.
```
Comment (kparichay) [inline Applications/MNIST/README.md:26]: In order to get the training data -> In order to feed the training data to the neural network model

Comment (kparichay) [inline Applications/MNIST/README.md:26]: If there is no data for an epoch -> If there is no `remaining` data for `the current` epoch

File: Applications/MNIST/README.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+``` c++
+/**
+ * @brief      get data which size is batch for train
+ * @param[out] outVec
```
Comment (kparichay) [inline Applications/MNIST/README.md:31]: Lets fill this variables description - can take from API header.

File: Applications/MNIST/README.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+ * @param[out] last if the data is finished
+ * @param[in] user_data private data for the callback
+ * @retval status for handling error
+ */
+int getBatch_train(float **outVec, float **outLabel, bool *last,
+                   void *user_data);
+```
+
+### MNIST Configuration file
+The model configuration is in ```res/mnist.ini```. ADAM optimizer is used and the trained model is going to be saved in ```model.bin```.
```
Comment (kparichay) [inline Applications/MNIST/README.md:42]: `model.bin` -> `./model.bin`?

File: Applications/MNIST/README.md [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+bias_initializer = zeros
+Activation = softmax
+
+```
+
+### How to run
+Once you compile the codes, you can run with
+
+``` bash
+$ ./nntrainer_mnist ${res}/mnist.ini
```
Comment (kparichay) [inline Applications/MNIST/README.md:114]: add `$ cd ${build_dir}`

File: Applications/MNIST/README.md (hunk 5 above)
Comment (kparichay) [reply Applications/MNIST/README.md:128]: Training_Kears -> Training_Keras

File: Applications/MNIST/README.md [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+
+``` bash
+$ python3 Training_Keras.py
+```
+
+```mnist_trainingSet.data``` must be in the same directory with ```Training_Kears.py```.
+
+
+### Comparison with Tensorflow
+This is the comparison with tensorflow-1.14.0 for the two cases. One is with zero initialization of weight and bias and the other is weight initialization data from tensorflow. As you can see, the results are pretty much same.
```
Comment (kparichay) [inline Applications/MNIST/README.md:132]: the other is weight initialization data from tensorflow -> the other is `random` weight initialization data `using the default intializers for each layer` from tensorflow

Comment (kparichay) [inline Applications/MNIST/README.md:132]: As you can see, the results are pretty much same -> As can be seen with the result below, the results are same within the margin of error.

File: Applications/MNIST/README.md (hunk 5 above)
Comment (jijoongmoon) [reply Applications/MNIST/README.md:128]: fixed. thanks.

File: Applications/MNIST/README.md (hunk 7 above)
Comment (jijoongmoon) [reply Applications/MNIST/README.md:31]: fixed. Thanks

File: Applications/MNIST/README.md (hunk 8 above)
Comment (jijoongmoon) [reply Applications/MNIST/README.md:42]: fixed. Thanks

File: Applications/MNIST/README.md (hunk 9 above)
Comment (jijoongmoon) [reply Applications/MNIST/README.md:114]: done.

File: Applications/MNIST/README.md (hunk 10 above)
Comment (jijoongmoon) [reply Applications/MNIST/README.md:132]: done.

Comment (jijoongmoon) [reply Applications/MNIST/README.md]: done. thanks.

**Review by kparichay — APPROVED** [review]
LGTM
